### PR TITLE
NXBT-1290: working image, tests, pipeline and README.md files

### PR DESCRIPTION
**Pull request by ffjdm**

### CODE REVIEW

**Review by pgmillon — APPROVED** [review]

**Comment by jcarsique** [comment]
:+1:

File: openldap/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Frantz FISCHER <ffischer@nuxeo.com>
+ */
+ 
+timeout(10) {
```
Comment (jcarsique) [inline openldap/Jenkinsfile:20]: timeout after the node allocation, else the time waiting for a node is counted

File: openldap/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    node('DYNAMIC') {
+        stage('checkout') {
+            checkout scm
```
Comment (jcarsique) [inline openldap/Jenkinsfile:23]: restrict the trigger based on directories or files using the Git regions:
```
checkout(
    [$class: 'GitSCM',
        branches: [[name: '*/${BRANCH}']],
        browser: [$class: 'GithubWeb', repoUrl: 'https://github.com/nuxeo/nuxeo'],
        doGenerateSubmoduleConfigurations: false,
        extensions: [
            [$class: 'PathRestriction', excludedRegions: '', includedRegions: '''nuxeo-distribution/.*integration/.*'''],
            [$class: 'SparseCheckoutPaths', sparseCheckoutPaths: [[path: 'nuxeo-distribution'], [path: 'integration']]],
            [$class: 'WipeWorkspace'],
            [$class: 'CleanBeforeCheckout'],
            [$class: 'CloneOption', depth: 5, noTags: true, reference: '', shallow: true]
        ],
        submoduleCfg: [],
        userRemoteConfigs: [[url: 'git://github.com/nuxeo/nuxeo.git']]
    ])
```

File: openldap/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            git_files_modified_for_repo = sh(returnStdout: true, script: 'git diff-tree --no-commit-id --name-only -r $(git rev-parse HEAD) | grep -c ^openldap || true').trim()
```
Comment (jcarsique) [inline openldap/Jenkinsfile:24]: `git diff-tree --no-commit-id --name-only -r HEAD -- openldap|wc -l`

File: openldap/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            echo "Number of files changed for this commit -> ${git_files_modified_for_repo}"
+        }
+        if (git_files_modified_for_repo == "0") {
```
Comment (jcarsique) [inline openldap/Jenkinsfile:27]: check after test if that bloc is needed when using the path restriction in the checkout step

File: openldap/Makefile [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+TAG-NAME=nuxeo/openldap
+NAME=dockerpriv.nuxeo.com:443/$(TAG-NAME)
+VERSION=1.0.0
+SCM-REF=PIPELINE-SCM-VALUE
+
+all: clean build test
+
+clean:
+	-docker-compose --file tests/docker-compose.yml rm -f
+	-docker rm $(docker ps --filter=status=exited --filter=status=created -q) 2>/dev/null
```
Comment (jcarsique) [inline openldap/Makefile:29]: Please add an echo after the first line to warn about the unexpected following cleanup

**Pull request merged**